.github/workflows/scripts/linux_arm64/build_catalyst.sh

```diff
@@ -0,0 +1,81 @@
+rm -rf /usr/bin/python3
+ln -s /opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/bin/python3 /usr/bin/python3 
+export PYTHON=/usr/bin/python3
+# Add LLVM, Python and GCC to the PATH env var
+export PATH=/catalyst/llvm-build/bin:/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/bin:/opt/rh/gcc-toolset-${GCC_VERSION}/root/usr/bin:$PATH
+# Install python dependencies
+/usr/bin/python3 -m pip install numpy pybind11 PyYAML cmake ninja delocate 'amazon-braket-pennylane-plugin>1.27.1'
+# Build Catalyst runtime
+cmake -S runtime -B runtime-build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=runtime-build/lib \
+    -DCMAKE_CXX_STANDARD_LIBRARIES="-lpython${PYTHON_VERSION}" \
+    -DPYTHON_EXECUTABLE=/usr/bin/python3 \
+    -DPYTHON_INCLUDE_DIR=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/include/python${PYTHON_VERSION} \
+    -DPYTHON_LIBRARY=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/lib \
+    -Dpybind11_DIR=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/lib/python${PYTHON_VERSION}/site-packages/pybind11/share/cmake/pybind11 \
+    -DENABLE_LIGHTNING_KOKKOS=ON \
+    -DLIGHTNING_GIT_TAG=latest_release \
+    -DENABLE_LAPACK=OFF \
+    -DKokkos_ENABLE_SERIAL=ON \
+    -DKokkos_ENABLE_OPENMP=ON \
+    -DKokkos_ENABLE_COMPLEX_ALIGN=OFF \
+    -DENABLE_WARNINGS=OFF \
+    -DENABLE_OPENQASM=ON \
+    -DENABLE_OPENMP=OFF \
+    -DLQ_ENABLE_KERNEL_OMP=OFF
+cmake --build runtime-build --target rt_capi rtd_lightning rtd_openqasm rtd_dummy
+# Patch OQC for correct linking
+echo "target_link_directories(rtd_oqc PRIVATE /catalyst/runtime-build/utils/runtime-build/lib)" >> frontend/catalyst/third_party/oqc/src/CMakeLists.txt
+export OQC_BUILD_DIR="/catalyst/oqc-build"
+export RT_BUILD_DIR="/catalyst/runtime-build"
+# Build OQC
+make oqc
+# Build Catalyst dialects
+cmake -S mlir -B quantum-build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DLLVM_ENABLE_ASSERTIONS=ON \
+    -DQUANTUM_ENABLE_BINDINGS_PYTHON=ON \
+    -DPython3_EXECUTABLE=/usr/bin/python3 \
+    -DPython3_NumPy_INCLUDE_DIRS=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/lib/python${PYTHON_VERSION}/site-packages/numpy/core/include \
+    -DMLIR_DIR=/catalyst/llvm-build/lib/cmake/mlir \
+    -DMHLO_DIR=/catalyst/mhlo-build/lib/cmake/mlir-hlo \
+    -DMHLO_BINARY_DIR=/catalyst/mhlo-build/bin \
+    -DEnzyme_DIR=/catalyst/enzyme-build \
+    -DENZYME_SRC_DIR=/catalyst/mlir/Enzyme \
+    -DLLVM_ENABLE_ZLIB=OFF \
+    -DLLVM_ENABLE_ZSTD=FORCE_ON \
+    -DLLVM_ENABLE_LLD=OFF \
+    -DLLVM_DIR=/catalyst/llvm-build/lib/cmake/llvm
+cmake --build quantum-build --target check-dialects compiler_driver
+# Copy files needed for the wheel where they are expected
+cp /catalyst/runtime-build/lib/*/*/*/*/librtd* /catalyst/runtime-build/lib
+cp /catalyst/runtime-build/lib/registry/runtime-build/lib/catalyst_callback_registry.cpython-39-aarch64-linux-gnu.so /catalyst/runtime-build/lib
```

Comment:
Also, is it expected to only have a cpython 3.9 module here? What happens with 3.10-3.12?